contrib/win32/win32compat/w32fd.c

```diff
@@ -1090,54 +1121,108 @@ spawn_child_internal(char* cmd, char *const argv[], HANDLE in, HANDLE out, HANDL
 	}
 
 	/* Add double quotes around the executable path
-	 * path can be c:\cygwin64\bin\sh.exe "<e:\openssh\regress\ssh-log-wrapper.sh>"
-	 * Please note that, this logic is not just bash test specific.
-	 */
+	* path can be c:\cygwin64\bin\sh.exe "<e:\openssh\regress\ssh-log-wrapper.sh>"
+	* Please note that, this logic is not just bash test specific.
+	*/
 	const char *exe_extenstion = ".exe";
 	const char *tmp = NULL;
 	if ((tmp = strstr(path, exe_extenstion)) && (strlen(tmp) > strlen(exe_extenstion))) {
-		tmp += strlen(exe_extenstion); /* move the pointer to the end of ".exe" */
-		
-		memcpy(t, path, strlen(path)-strlen(tmp));
-		t += strlen(path) - strlen(tmp);
-
-		*t++ = '\"';
-		memcpy(t, tmp, strlen(tmp));
-		t += strlen(tmp);
-	} else {
+		tmp += strlen(exe_extenstion); /* move the pointer to the end of ".exe" */		
```

Comment:
can you end up with an input command like the following here:

  c:\windows\system32\cmd /c ping.exe localhost

If so, this logic would incorrectly identify the nested binary as the parent executable

Reply:
For the existing known shells and ssh executable, the first parts always ends with .exe. I don't see other ways to identify the file path (the first part)  in the string.

Reply:
What about with proxyCommand or SSH_ASK_PASS ?

Reply:
Yes, you are right.

Reply:
in next version, I use the last parameter to define if it is user defined cmd. If so, we don't change anything. we expect the cmd user enter already have right quotes and escaping